projects/core/src/occ/config-loader/occ-config-loader.module.ts

```diff
@@ -12,8 +13,15 @@ import { OccConfigLoaderService } from './occ-config-loader.service';
  */
 export function initConfig(
   configLoader: OccConfigLoaderService,
-  config: SiteContextConfig
-): ConfigInitializer {
+  config: SiteContextConfig,
+  /**
+   * TODO: remove it in 4.0
+   */
+  featureConfigService?: FeatureConfigService
```

Comment:
This can be made mandatory. `initConfig` function is not in public API. no breaking change

Reply:
We will leave this `  featureConfigService?: FeatureConfigService` as mandatory? Why we still need it in 4.0?

Reply:
I'd say it doesn't matter whether it's mandatory or not. I'm just saying it _doesn't need to_ be optional, because it was not in Public API before, so introducing a new required param is not a breaking change.

We're going to drop the whole factory function in 4.0 anyway ;)